### ISSUE TITLE
2 messages one after the other does not seem to work, needs some waiting.

### ISSUE THREAD

**Issue by hoekbrwr** [Links2004/arduinoWebSockets#531]
I found out that that 2 messages sent directly after each other overwrite each other. I tried with broadcastTXT and with sendTXT and first tried 20msec then at 200msec, there is no collision. Is there a way to find out if the first message is ready and the next message can be sent?

**Comment by Links2004** [comment]
can you post a code example?
the only way I can think of how this is possible is when you override the payload pointer that you give as parameter,
or the TCP stack has a bug, but the code is normally designed in a way that you can fire multible calls on after a another (until the TCP stack is full).

**Comment by hoekbrwr** [comment]
What I do is sending out a position some sliders are in. Then with the app I send I can adjust the same sliders to another setting. So at themoment you connext with the app you have the actual setting to start with. So I send the messages directly after I got the message the app is connected to the websocket server. This is a piece of code the same as you use in your examples:
`void webSocketEvent(uint8_t num, WStype_t type, uint8_t *payload, size_t length)
{ // When a wsServer message is received
  switch (type)
  {
  case WStype_DISCONNECTED: // if the wsServer is disconnected
    Serial.printf("[%u] Disconnected!\n", num);
    break;
  case WStype_CONNECTED:
  { // if a new wsServer connection is established
    IPAddress ip = wsServer.remoteIP(num);
    Serial.printf("[%u] Connected from %d.%d.%d.%d url: %s\n", num, ip[0], ip[1], ip[2], ip[3], payload);
    wsServer.sendTXT(num,"H" + sliderHokValue);
    delay(200); //Blijkbaar overschrijft volgende als je niet even wacht tot de vorige klaar is
                // Nog uitzoeken of er een readymelding komt
    wsServer.sendTXT(num,"R" + sliderRenValue);
  }
  break;`
I did not suspect this could happen because websockets are so fast.

**Comment by Links2004** [comment]
you are accessing some random memory addr with:
```
"H" + sliderHokValue
```
`"H"` is a `const char *` as a result you are moving the addr ptr in the RAM with what value is stored in `sliderHokValue`, as result you send what is in that new memory location.

if you want arduino like String addition you need to use:

```
String("H") + sliderHokValue
```

**Comment by Links2004** [comment]
have done some testing, the lib is working fine.

```cpp
        case WStype_CONNECTED: {
            IPAddress ip = webSocket.remoteIP(num);
            USE_SERIAL.printf("[%u] Connected from %d.%d.%d.%d url: %s\n", num, ip[0], ip[1], ip[2], ip[3], payload);

            // send message to client
            webSocket.sendTXT(num, "Connected");

            webSocket.sendTXT(num, "Connected 2");
            webSocket.sendTXT(num, "Connected 3");
            webSocket.sendTXT(num, "Connected 4");
        }

```

![Screenshot from 2020-04-27 11-47-13](https://user-images.githubusercontent.com/974709/80358583-f1aa1800-887c-11ea-944c-ca8c0a973019.png)

**Comment by hoekbrwr** [comment]
Thanks for your investigation. I investigated it a little further and concluded it is not the arduinoWebsockets library who is to blame. I remembered also your RGB demo where I've been seeing flouds of colour messages. I also have a webserver page and when I connect that to the server there is no lost message! 
I tried your suggestion to use String() , but that did not make any difference. 
The problem is in the Flutter/Dart widget losing the first message. My webpage is receiving all messages!
